pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -429,11 +429,22 @@ func readSavedProtoMessage(staleConfigTime uint32,
 // The most recent config hash we received. Starts empty
 var prevConfigHash string
 
-func generateConfigRequest() ([]byte, *zconfig.ConfigRequest, error) {
+func generateConfigRequest(getconfigCtx *getconfigContext) ([]byte, *zconfig.ConfigRequest, error) {
 	log.Debugf("generateConfigRequest() sending hash %s", prevConfigHash)
 	configRequest := &zconfig.ConfigRequest{
 		ConfigHash: prevConfigHash,
 	}
+	iToken, err := readIntegrityToken()
+	if err != nil {
+		//log an error, but proceed with request.
+		//if controller rejects config request because of
+		//missing integrity token, attestation will be
+		//retriggered, which will again repopulate a fresh
+		//integrity token
+		log.Errorf("Failed to read integrity token. err: %v", err)
```

Comment:
Will we always hit this the first time a device is installed or booted i.e., noice error messages? Or are we waiting to send the first config request until we have completed attestation?

Reply:
> Will we always hit this the first time a device is installed or booted i.e., noice error messages? Or are we waiting to send the first config request until we have completed attestation?

Currently attestation and config fetch tasks run independent of each other. So yes, for the first time (or till there is a successful attestation after reboot), this error will be there. IMHO, this is trade-off is ok compared to additional complexity involved in co-ordinating between these two tasks. Please let me know what you think.

Reply:
OK, good to know we might see some errors on startup.

But I wonder whether you should just silently treat it as an empty string when the read fails (or when it fails because the file does not exist).

Reply:
> OK, good to know we might see some errors on startup.
> 
> But I wonder whether you should just silently treat it as an empty string when the read fails (or when it fails because the file does not exist).
@eriknordmark I've updated the changes, please take a look. We now populate integrity token(whatever we get from the file) when read is successful, and we write to the file even if we get an empty or nil value. I've removed the error message to align with this logic.

Reply:
> OK, good to know we might see some errors on startup.
> 
> But I wonder whether you should just silently treat it as an empty string when the read fails (or when it fails because the file does not exist).

@eriknordmark I've updated the changes, please take a look. We now populate integrity token(whatever we get from the file) when read is successful, and we write to the file even if we get an empty or nil value. I've removed the error message to align with this logic.